src/solvers/dg_structured/dg_3d.jl

```diff
@@ -0,0 +1,292 @@
+    for j in eachnode(dg), i in eachnode(dg), v in eachvariable(equations)
+      element.interfaces[5].u_right[v, i, j] = u[v, i, j, 1, element_ind]
+      element.interfaces[6].u_left[v, i, j] = u[v, i, j, end, element_ind]
+    end
+  end
+
+  return nothing
+end
+
+
+function prolong2boundaries!(cache, u::AbstractArray{<:Any,5}, 
+    boundary_condition::BoundaryConditionPeriodic, mesh::StructuredMesh, equations, dg::DG)
+  @unpack size, linear_indices = mesh
+
+  # Boundaries in x-direction
+  for element_y in 1:size[2], element_z in 1:size[3]
+    @views cache.elements[1, element_y, element_z].interfaces[1].u_left .= u[:, end, :, :, linear_indices[end, element_y, element_z]]
+    @views cache.elements[end, element_y, element_z].interfaces[2].u_right .= u[:, 1, :, :, linear_indices[1, element_y, element_z]]
+  end
+
+  # Boundaries in y-direction
+  for element_x in 1:size[1], element_z in 1:size[3]
+    @views cache.elements[element_x, 1, element_z].interfaces[3].u_left .= u[:, :, end, :, linear_indices[element_x, end, element_z]]
+    @views cache.elements[element_x, end, element_z].interfaces[4].u_right .= u[:, :, 1, :, linear_indices[element_x, 1, element_z]]
+  end
+
+  # Boundaries in z-direction
+  for element_x in 1:size[1], element_y in 1:size[2]
+    @views cache.elements[element_x, element_y, 1].interfaces[5].u_left .= u[:, :, :, end, linear_indices[element_x, element_y, end]]
+    @views cache.elements[element_x, element_y, end].interfaces[6].u_right .= u[:, :, :, 1, linear_indices[element_x, element_y, 1]]
+  end
+
+  return nothing
+end
+
+
+function calc_interface_flux!(nonconservative_terms::Val{false}, mesh::StructuredMesh{3}, equations,
+                              dg::DG, cache)
+  @unpack surface_flux = dg
+  @unpack size = mesh
+
+  @threaded for element in eachelement(dg, cache)
+    # Interfaces in negative directions
+    for orientation in (1, 3, 5)
+      interface = cache.elements[element].interfaces[orientation]
+      calc_interface_flux!(interface, mesh, equations, dg)
+    end
+  end
+
+  # Boundary in positive x-direction
+  # TODO maybe make this threaded with @threaded for element in @view(mesh.linear_indices[end, :, :])
+  for element_y in 1:size[2], element_z in 1:size[3]
+    interface = cache.elements[end, element_y, element_z].interfaces[2]
+    calc_interface_flux!(interface, mesh, equations, dg)
+  end
+
+  # Boundary in positive y-direction
+  for element_x in 1:size[1], element_z in 1:size[3]
+    interface = cache.elements[element_x, end, element_z].interfaces[4]
+    calc_interface_flux!(interface, mesh, equations, dg)
+  end
+
+  # Boundary in positive z-direction
+  for element_x in 1:size[1], element_y in 1:size[2]
+    interface = cache.elements[element_x, element_y, end].interfaces[6]
+    calc_interface_flux!(interface, mesh, equations, dg)
+  end
+
+  return nothing
+end
+
+
+function calc_interface_flux!(interface::Interface, mesh::StructuredMesh{3}, equations, dg::DG)
```

Comment:
```suggestion
function calc_interface_flux!(interface, mesh::StructuredMesh{3}, equations, dg::DG)
```
AFAICT, the type annotation is not needed for dispatch, in which case I'd prefer not to have it (unless you know that there are going to be other kinds of interfaces)